### ISSUE TITLE
If a tsc command is not able to run because there is no tsconfig.json file give the user some sort of hint about that

### ISSUE THREAD

**Issue by jcollum-nutrien** [microsoft/TypeScript#50615]
# Suggestion

## 🔍 Search Terms

tsconfig.json

## ✅ Viability Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
* [?] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

## ⭐ Suggestion

Here's what I did: 

```
✗ cat package.json | grep type
    "typecheck": "tsc --noEmit",
    "typescript": "~4.8.2",

✗ npx which tsc
/Users/justin.collum2@nutrien.com/work/recommendations-dp/node_modules/.bin/tsc

✗ yarn typecheck
```

Result: I get the help text, which implies that I used an invalid CLI flag. But the CLI flag I used is in the help text! So it was very confusing. 

Then I realized that this is a monorepo and it doesn't have a root tsconfig file: 

```
✗ ls -alh | grep tsconfig

✗
```

I suspect that's the issue. 

If `tsc` can't proceed because there is no config file that it is able to find, say so. Don't emit the help text, emit an error like "Can't proceed, no config file found, search paths: ..." 

Just getting the CLI help text was very misleading.

**Comment by fatcerberus** [comment]
It's possible to run `tsc` without a tsconfig, e.g.

```
  tsc app.ts util.ts
  Ignoring tsconfig.json, compiles the specified files with default compiler options.
```

So it's not the case that "tsc can't proceed because there is no config file" - it can't proceed because you haven't given it *any* input at all and there's no tsconfig for it to fall back on.  I agree this fact should be communicated better, though.

**Comment by jcollum-nutrien** [comment]
OK In that case I'd expect an error like "no config and no inputs so the command line flag is ignored" etc.

**Comment by nraychaudhuri** [comment]
I want to work on this issue. Find a way to provide a better error message.

**Comment by kamweti** [comment]
@jcollum-nutrien

looks like the `tsc` command without any input or flags is used as a synonym for `tsc --help`, when a `tsconfig.json` is not found in the working directory.

changing this behaviour to instead show an error is a breaking change in my opinion

use the `--showConfig` flag instead which provides a hint of missing config

```
tsc --showConfig
> error TS5081: Cannot find a tsconfig.json file at the current directory: ~/Projects/foo
```

**Comment by ValerianClerc** [comment]
@kamweti I agree that returning an error code to the shell from that warning could potentially be a breaking change. But we could inform users with a simple informational log to the console?
